spec/models/king_spec.rb

```diff
@@ -94,4 +94,74 @@
       expect(moved_king.moved).to eq true
     end
   end
+
+  describe 'castling' do
+    let(:castling_king) do
+      empty_game.pieces.create(
+        type: 'King',
+        color: 'white',
+        x_position: 4,
+        y_position: 7
+      )
+    end
+    let(:castling_rook) do
+      empty_game.pieces.create(
+        type: 'Rook',
+        color: 'white',
+        x_position: 7,
+        y_position: 7
+      )
+    end
+    let(:obstruction) do
+      empty_game.pieces.create(
+        type: 'Bishop',
+        color: 'white',
+        x_position: 1,
+        y_position: 7
+      )
+    end
+    let(:obstructed_rook) do
+      empty_game.pieces.create(
+        type: 'Rook',
+        color: 'white',
+        x_position: 0,
+        y_position: 7
+      )
+    end
+
+    it 'should return true on valid castling' do
+      castling_rook.save
+      expect(castling_king.move!(6, 7)).to eq true
+      expect(castling_king.x_position).to eq 6
+      expect(castling_king.y_position).to eq 7
+      expect(castling_king.moved).to eq true
+      castling_rook.reload
+      expect(castling_rook.x_position).to eq 5
+      expect(castling_rook.y_position).to eq 7
+      expect(castling_rook.moved).to eq true
+    end
+
+    it 'should return false on obstructed castling' do
+      obstructed_rook.save
+      obstruction.save
+      expect(castling_king.move!(2, 7)).to eq false
+      expect(castling_king.x_position).to eq 4
+      expect(castling_king.y_position).to eq 7
+      expect(castling_king.moved).to eq false
+      expect(obstructed_rook.x_position).to eq 0
+      expect(obstructed_rook.y_position).to eq 7
+      expect(obstructed_rook.moved).to eq false
+    end
+
+    it 'should return false on castling when rook moved' do
+      castling_rook.update(moved: true)
+      expect(castling_king.move!(6, 7)).to eq false
+      expect(castling_king.x_position).to eq 4
+      expect(castling_king.y_position).to eq 7
+      expect(castling_king.moved).to eq false
+      expect(castling_rook.x_position).to eq 7
+      expect(castling_rook.y_position).to eq 7
+      expect(castling_rook.moved).to eq true
+    end
+  end
```

Comment:
Yeah, I was naughty and didn't give full coverage because I was trying to do it all in one pass in a separate PR where I reinstate FactoryGirl for all tests.  However, I am ashamed and will add tests for black coverage.